common/define-grammar.js

```diff
@@ -881,7 +892,7 @@ module.exports = function defineGrammar(dialect) {
         field('type_parameters', optional($.type_parameters)),
         field('parameters', $.formal_parameters),
         field('return_type', optional(
-          choice($.type_annotation, $.asserts, $.type_predicate_annotation)
+          choice($.type_annotation, $.asserts_annotation, $.type_predicate_annotation)
         ))
```

Comment:
Correct me if I'm wrong, but asserts is not exactly a return_type, maybe make it a choice between `asserts_annotation` and `field('return_type', ...)`?

Reply:
Well, I would say that it is just as much a return type as a type predicate annotation.
```typescript
function f (x: unknown): string {}      // regular return type
function g (x: unknown): x is string {} // type predicate, it actually returns a boolean
function h (x: unknown): asserts x {}   // assertion, it actually returns void (or throws)
```

Reply:
I'm not super familiar with typescript compared to javascript - do these actually fit under return types? Or is it just an annotation

Reply:
You can see a type predicate signature as a fancier version of returning a boolean. It means that the function returns a boolean, but that in addition calling it will narrow the type of a variable in the calling scope. Similarly, assertion signatures are a fancier version of returning void. They return void, and also narrow the type of a variable. You can see [here](https://www.typescriptlang.org/play?#code/CYUwxgNghgTiAEYD2A7AzgF3gMwFzwAoAPfAVxQGsUkB3FASngF4A+eI+ASzXkxk5QBzANwBYAFChIsBMnRZB+YmUrU6jVvCho0IGBh5Ex4iXMzwAngEZmOAgGZ6xgPTP47+AD14AIyRIIECgUU1RzCwAmW0ECACIiWKcJVw8veAA3JE5gIA) that this is how Typescript interprets the return type. In particular, it really replaces the return type annotation and cannot be combined with another return type for instance.

Reply:
Thanks, that was insightful (especially w/ the demo)